app/src/components/__tests__/TeamContracts.spec.ts

```diff
@@ -0,0 +1,278 @@
+import { mount } from '@vue/test-utils'
+import { describe, it, expect, vi, beforeEach } from 'vitest'
+import TeamContracts from '@/components/TeamContracts.vue'
+import ModalComponent from '@/components/ModalComponent.vue'
+import TeamContractAdmins from '@/components/TeamContractAdmins.vue'
+import TeamContractsDetail from '@/components/TeamContractsDetail.vue'
+import { createPinia, setActivePinia } from 'pinia'
+
+const getAdminListMock = vi.fn()
+const removeAdminMock = vi.fn()
+const getEventsGroupedByCampaignCodeMock = vi.fn()
+// Mock AddCampaignService methods
+vi.mock('@/services/AddCampaignService', () => ({
+  AddCampaignService: vi.fn().mockImplementation(() => ({
+    getContractData: vi.fn().mockResolvedValue([{ key: 'costPerClick', value: '10' }]),
+    getAdminList: getAdminListMock,
+    getEventsGroupedByCampaignCode: getEventsGroupedByCampaignCodeMock,
+    removeAdmin: removeAdminMock
+  }))
+}))
+
+describe('TeamContracts.vue', () => {
+  const contracts = [
+    {
+      type: 'Campaign',
+      address: '0x1234567890abcdef1234567890abcdef12345678',
+      admins: ['0xadmin1Address', '0xadmin2Address'],
+      deployer: '0xdeployer1Address'
+    },
+    {
+      type: 'Campaign',
+      address: '0xabcdefabcdefabcdefabcdefabcdefabcdefabcd',
+      admins: ['0xadmin3Address'],
+      deployer: '0xdeployer1Address'
+    }
+  ]
+
+  beforeEach(() => {
+    const pinia = createPinia()
+    setActivePinia(pinia)
+  })
+
+  it('renders contract rows correctly', () => {
+    const wrapper = mount(TeamContracts, {
+      props: { contracts, teamId: 'team1' },
+      global: {
+        plugins: [createPinia()]
+      }
+    })
+
+    // Check that the rows match the number of contracts
+    const rows = wrapper.findAll('tbody tr')
+    expect(rows.length).toBe(contracts.length)
+
+    // Check that each row contains the correct contract data
+    rows.forEach((row, index) => {
+      const cells = row.findAll('td')
+
+      expect(cells[0].text()).toBe(contracts[index].type)
+      expect(cells[1].text()).toBe(contracts[index].address)
+      expect(cells[2].text()).toContain(
+        contracts[index].address.slice(0, 6) + '...' + contracts[index].address.slice(-4)
+      )
+    })
+  })
+
+  it('opens modal with contract details when a row is clicked', async () => {
+    const wrapper = mount(TeamContracts, {
+      props: { contracts, teamId: 'team1' },
+      global: {
+        plugins: [createPinia()]
+      }
+    })
+
+    // Simulate clicking the first row
+    const firstRow = wrapper.find('tbody tr')
+    const firstRowCells = firstRow.findAll('td')
+    await firstRowCells[4].trigger('click')
```

Comment:
You can simply use data test attribute as selector here, so you won't have error when the component place or position change in the future.